data/model/repository.py

```diff
@@ -739,17 +739,21 @@ def force_cache_repo_size(repo_id: int):
     except Manifest.DoesNotExist:
         size = 0
 
-    repo = get_repository_size(repo_id)
-
     if size is None:
         size = 0
 
-    if repo is not None:
-        update = RepositorySize.update(size_bytes=size).where(
-            RepositorySize.repository_id == repo_id
-        )
-        update.execute()
-    else:
-        RepositorySize.create(repository_id=repo_id, size_bytes=size)
+    with db_transaction():
+        repo_size_ref = get_repository_size(repo_id)
+        try:
+            if repo_size_ref is not None:
+                update = RepositorySize.update(size_bytes=size).where(
+                    RepositorySize.repository_id == repo_id
+                )
+                update.execute()
+            else:
+                RepositorySize.create(repository_id=repo_id, size_bytes=size)
+        except IntegrityError:
```

Comment:
can we add a comment here to explain that the only known way to reach this expection is due to race condition and the assumed behavior is that both with have the same value to cache?

Reply:
Done.